.github/workflows/codecov.yml

```diff
@@ -0,0 +1,63 @@
+name: Generate Codecov Report
+
+on:
+  push:
+    branches: [ develop, main ]
+  pull_request:
+    branches: [ develop, main ]
+
+jobs:
+  codecov_report:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        ref: ${{ github.head_ref }}
+        fetch-depth: 0
+          
+    - name: set up JDK 11
+      uses: actions/setup-java@v2
+      with:
+        java-version: '11'
+        distribution: 'adopt'
+        cache: gradle
+
+    - name: Grant execute permission for gradlew
```

Comment:
Use the Gradle build action as is done [here](https://github.com/newrelic/newrelic-android-agent/blob/develop/.github/workflows/functionalTests.yml#L22-L25).